lib/saluki-components/src/sources/checks/scheduler.rs

```diff
@@ -54,6 +55,22 @@ impl Scheduler {
         let check_id = check.id();
         let interval_secs = check.interval().as_secs();
 
+        if interval_secs == 0 {
+            // Check with no interval are push to a random worker immediately
+            let channels_guard = self.worker_channels.lock().unwrap();
+            if channels_guard.is_empty() {
+                warn!("Failed to schedule one-time check '{}': No workers available", check_id);
+                return;
+            }
+
+            let channel_idx = rand::thread_rng().gen_range(0..channels_guard.len());
+            if let Err(e) = channels_guard[channel_idx].try_send(WorkerMessage::RunCheck(Arc::clone(&check))) {
+                error!("Failed to enqueue a one-time check '{}': {}", check_id, e);
+            }
+            info!("Scheduled one-time check '{}' to worker {}", check_id, channel_idx);
```

Comment:
```suggestion
            info!(check_id, "Scheduled one-time check.");
```

Similar comment about structured key/value pairs as made below.

As far as not mentioning the worker index, I would say that we ought to just have each worker emit a log message about each check as they execute it. As is, if it's not relevant to mention the worker index for a scheduled check, doesn't seem worth it to mention it for a one-time check either? (This isn't a strongly held opinion but I think it ends up a little cleaner to just have workers log when they're running a check then the schedule logic itself emitting that.)